src/groupederror.jl

```diff
@@ -202,14 +204,24 @@ Output is a `GroupedError` with error computed according to the keyword `compute
 It can be plotted using `plot(g::GroupedError)`
 Seriestype can be specified to be `:path`, `:scatter` or `:bar`
 """
-function groupapply(f::Function, df, args...;
-                    axis_type = :auto, compute_error = :none, group = [],
+function groupapply(f::Function, dfn, args...;
+                    axis_type = :auto, compute_error = :none, group = Array{Symbol,1}(0),
                     summarize = (get_symbol(compute_error) == :bootstrap) ? (mean, std) : (mean, sem),
                     compute_axis = :separate,
                     kwargs...)
+
     if !(axis_type in [:discrete, :continuous])
-        axis_type = (typeof(df[args[1]])<:PooledDataArray) ? :discrete : :continuous
+        axis_type = (typeof(dfn[args[1]])<:Union{CategoricalArray,NullableCategoricalArray}) ?
+                    :discrete : :continuous
     end
+
+    # Convert to dataframe with only relevant columns and no missing data
+    err_col = (isa(compute_error, Tuple) && (compute_error[1] == :across)) ?
+                vectorify(compute_error[2]) : Array{Symbol,1}(0)
+    rel_cols = filter(t -> (t in names(dfn)), union([args...], err_col, vectorify(group)))
+    df = dfn[complete_cases(dfn[:, rel_cols]),rel_cols]
+    df = DataFrame(convert.([Array],getindex.([df],rel_cols)) , rel_cols)
```

Comment:
`df[rel_cols]` won't work, because you need an array of columns for the dot syntax to work, whereas `df[rel_cols]` returns a dataframe. Then the array of column becomes an array of arrays, which is the only way to get a dataframe without nullable arrays AFAIK. `Array.(...)` is much more legible than what I have and it does just the same, I'll change to that.